### PR TITLE
CNS-61 feat: leave room, improve roomModal, text size

### PR DESCRIPTION
**Pull request by yun-si**
## Changes
- 邀請碼輸入
  - 改成大小寫不敏感
  - 再點開的時候可以自動把之前的輸入清掉
- 離開房間的機制
- 修改 text size

## Test
1. follow .env.example in dev branch to set up .env
2. start frontend and backend

### (1) lowercase is allowed
image

**join room success**
image

### (2) input is cleared
image

### (3) leave room

**before leaving room**
image

**after leaving room**
image

**roommate has left**
image

### CODE REVIEW

File: backend/src/controllers/roomsController.js [hunk 1]
```diff
@@ -83,6 +83,35 @@ class RoomsController {
...
+        logWithFileInfo('info', '[RoomsController] -----leaveTargetRoom-----');
+
+        let user;
+        let token;
+        try {
+            user = req.body.user;
+            token = req.params.roomToken;
+
+            if (!user) {
+                logWithFileInfo('error', '[RoomsController] Error when joining room - User object is required');
```
Comment (JLin056) [inline backend/src/controllers/roomsController.js]: 打錯了～ when leaving room

File: backend/src/controllers/roomsController.js [hunk 2]
```diff
@@ -83,6 +83,35 @@ class RoomsController {
...
+                return res.status(400).json({ message: 'User object is required' });
+            }
+
+            if (!user.id || user.id.length === 0) {
+                logWithFileInfo('error', '[RoomsController] Error when joining room - User id is required');
```
Comment (JLin056) [inline backend/src/controllers/roomsController.js]: 打錯了～ when leaving room

**Review by JLin056 — COMMENTED** [review]
前端沒問題～
剩 log 改一下就好！

**Review by JLin056 — DISMISSED** [review]
great

**Review by amyhsu-o — APPROVED** [review]
待調整：重新加入房間後的聊天室訊息 bug

**Pull request merged**